homeassistant/components/tellstick.py

```diff
@@ -59,12 +59,12 @@ def _discover(hass, config, component_name, found_tellcore_devices):
 def setup(hass, config):
     """Setup the Tellstick component."""
     from tellcore.constants import TELLSTICK_DIM
-    from tellcore.library import DirectCallbackDispatcher
+    from tellcore.telldus import AsyncioCallbackDispatcher
     from tellcore.telldus import TelldusCore
 
     try:
         tellcore_lib = TelldusCore(
-            callback_dispatcher=DirectCallbackDispatcher())
+            callback_dispatcher=AsyncioCallbackDispatcher(hass.loop))
```

Comment:
I'm still to green on Python asyncio to answer about the coroutines, but the effect is that handling of callbacks from the telldus native library is transferred using hass.loop.call_soon_threadsafe(). The old code had the actual callback executing in the thread of the native telldus library.

It's possible that I can turn _tellcore_event_callback() and/or update_from_callback() into coroutines, but I'm not sure how to do that, or what the effect/benefit of that would be.